src/iidm/ThreeWindingsTransformerAdder.cpp

```diff
@@ -7,161 +7,42 @@
-    m_voltageLevelId.clear();
-    m_node.reset();
-    m_bus.clear();
-    m_connectableBus.clear();
-    m_r = stdcxx::nan();
-    m_x = stdcxx::nan();
-    m_g = stdcxx::nan();
-    m_b = stdcxx::nan();
-    m_ratedU = stdcxx::nan();
-
-    return *this;
-}
-
-std::string ThreeWindingsTransformerAdder::LegAdder::getMessageHeader() const {
-    static std::array<std::string, 3> s_messageHeaders {{
-        u8"3 windings transformer leg1: ",
-        u8"3 windings transformer leg2: ",
-        u8"3 windings transformer leg3: "
-    }};
-
-    return s_messageHeaders[m_legNumber - 1];
-}
-
-ThreeWindingsTransformerAdder::LegAdder& ThreeWindingsTransformerAdder::LegAdder::setB(double b) {
-    m_b = b;
-    return *this;
-}
-
-ThreeWindingsTransformerAdder::LegAdder& ThreeWindingsTransformerAdder::LegAdder::setBus(const std::string& bus) {
-    m_bus = bus;
-    return *this;
-}
-
-ThreeWindingsTransformerAdder::LegAdder& ThreeWindingsTransformerAdder::LegAdder::setConnectableBus(const std::string& connectableBus) {
-    m_connectableBus = connectableBus;
-    return *this;
-}
-
-ThreeWindingsTransformerAdder::LegAdder& ThreeWindingsTransformerAdder::LegAdder::setG(double g) {
-    m_g = g;
-    return *this;
-}
-
-ThreeWindingsTransformerAdder::LegAdder& ThreeWindingsTransformerAdder::LegAdder::setNode(unsigned long node) {
-    m_node = node;
-    return *this;
-}
-
-ThreeWindingsTransformerAdder::LegAdder& ThreeWindingsTransformerAdder::LegAdder::setR(double r) {
-    m_r = r;
-    return *this;
-}
-
-ThreeWindingsTransformerAdder::LegAdder& ThreeWindingsTransformerAdder::LegAdder::setRatedS(double ratedS) {
-    m_ratedS = ratedS;
-    return *this;
-}
-
-ThreeWindingsTransformerAdder::LegAdder& ThreeWindingsTransformerAdder::LegAdder::setRatedU(double ratedU) {
-    m_ratedU = ratedU;
-    return *this;
-}
-
-ThreeWindingsTransformerAdder::LegAdder& ThreeWindingsTransformerAdder::LegAdder::setVoltageLevel(const std::string& voltageLevelId) {
-    m_voltageLevelId = voltageLevelId;
-    return *this;
-}
-
-ThreeWindingsTransformerAdder::LegAdder& ThreeWindingsTransformerAdder::LegAdder::setX(double x) {
-    m_x = x;
-    return *this;
-}
-
 ThreeWindingsTransformerAdder::ThreeWindingsTransformerAdder(Substation& substation) :
-    m_substation(substation),
-    m_adder1(*this, 1),
-    m_adder2(*this, 2),
-    m_adder3(*this, 3) {
+    m_substation(substation) {
 }
 
 ThreeWindingsTransformer& ThreeWindingsTransformerAdder::add() {
     logging::Logger& logger = logging::LoggerFactory::getLogger<ThreeWindingsTransformer>();
 
-    std::unique_ptr<ThreeWindingsTransformer::Leg> ptrLeg1 = m_adder1.checkAndGetLeg();
-    VoltageLevel& voltageLevel1 = m_adder1.checkAndGetVoltageLevel();
-    std::unique_ptr<Terminal> ptrTerminal1 = m_adder1.checkAndGetTerminal(voltageLevel1);
+    if (!m_adder1.is_initialized()) {
+        throw ValidationException(newLeg1(), "leg 1 is not defined");
```

Comment:
This is not a good idea to create a LegAdder to get a validable. Use `*this` instead